### ISSUE TITLE
Allow disabling `Lint/Syntax` for files/methods that are not strictly Ruby

### ISSUE THREAD

**Issue by maedi** [rubocop/rubocop#14999]
### Before submitting

- [x] I have searched the existing issues and this hasn't been requested yet.

### Cop name

Lint/Syntax

### Is your feature request related to a problem?

I'm doing something [weird](https://github.com/low-rb/low_node):
```ruby
class MockNode < LowNode
  def render
    <p>Hello</p>
  end
end
```

And loading it up via an [autoloader](https://github.com/low-rb/lowload) that automatically [wraps](https://github.com/low-rb/lowkey) the invalid code into a valid string. It's really not weird considering other languages embed HTML into their code (JSX)... albeit via transpilation, but that's more of a technical detail; this file is still valid Ruby by the time it's loaded, but RuboCop says:
```
spec/fixtures/mock_node.rbx:5:14: F: Lint/Syntax: unterminated string meets end of file
(Using Ruby 3.3 parser; configure using TargetRubyVersion parameter, under AllCops)
    <p>Hello</p>
```

I would prefer not to disable RuboCop entirely per file extension:
```yaml
AllCops:
  Exclude:
    - '**/*.rbx'
```

And it's not possible to disable the `Lint/Syntax` cop for this file extension:
```yaml
Lint/Syntax:
  Exclude:
    - '*.rbx'
```
...which results in:
```
Error: configuration for Lint/Syntax cop found in .rubocop.yml
It's not possible to disable this cop.
```

It's interesting that RuboCop can validate an `.rbx` file in the first place?

Related issue: https://github.com/rubocop/rubocop/issues/10509

### Describe the solution you'd like

Please allow libraries that manipulate Ruby code before it's loaded to disable syntax linting for the file/method that they control:

Per file:
```yaml
Lint/Syntax:
  Exclude:
    - '*.rbx'
```

Ideally per file type/method name:
```yaml
Lint/Syntax:
  Exclude:
    - '*.rbx#render'
```

**Comment by bbatsov** [comment]
The `.rbx` extension in RuboCop's default configuration is a historical artifact from [Rubinius](https://en.wikipedia.org/wiki/Rubinius), an alternative Ruby implementation that's been effectively dead since ~2020. It's listed both as a `RubyInterpreter` and in `AllCops: Include` in our `default.yml`. It has nothing to do with your use case of embedding HTML in Ruby files.

Since your `.rbx` files aren't actually Ruby, `AllCops: Exclude` is the correct solution:

```yaml
AllCops:
  Exclude:
    - '**/*.rbx'
```

There's no practical benefit to allowing `Lint/Syntax` exclusions separately — if a file can't be parsed, no other cops can meaningfully run on it either.

That said, we should probably remove `.rbx` from the default `Include` list entirely, since Rubinius is no longer maintained and nobody is writing new Rubinius-specific files. I'll look into that as a separate cleanup.
